### PR TITLE
Electron Mac file associations

### PR DESCRIPTION
**Pull request by timtmok**
### Intent
Addresses #10893

Allows opening associated files, dragging files on to the app icon, and command-line arguments

### Approach
Feeds in the `Info.plist.in` template when building the app. This associates the files with MacOS.

Listens to the `open-file` event which triggers for associated file open events and dragging files on to the app icon. For project files, this sets the env var before the session is created so that the project is opened.

For command-line arguments, they are searched for unswitched options, checks if it's a valid file, and opens it.

An `initialized` property was added to the GWT callback so it can track when GWT has initialized the workbench. Responding to the app open event needs to know if the workbench is ready to call the open file action.

### Automated Tests
None

### QA Notes
Main cases to test:
* dragging files on to the app icon (in Finder or the dock)
* double-clicking files in Finder
* opening from the command-line (`open RStudio.app --args /path/to/file`)
* opening a second project should open a new instance of RStudio

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/node/desktop/forge.config.js [hunk 1]
```diff
@@ -46,6 +46,7 @@ const config = {
   packagerConfig: {
     icon: './resources/icons/RStudio',
     appBundleId: 'com.rstudio.desktop',
+    extendInfo: '../../cpp/desktop/Info.plist.in',
```
Comment (gtritchie) [inline src/node/desktop/forge.config.js:49]: We can't directly use this file, it needs to be consumed after cmake has generated the output file (Info.plist) via [configure_file](https://cmake.org/cmake/help/latest/command/configure_file.html#command:configure_file). Currently this doesn't happen for the Electron build (it's in the Qt-desktop branch of the cmakefile): https://github.com/rstudio/rstudio/blob/88d5f9b1359d261f56a8721ca1bc03d13fbc9701/src/cpp/desktop/CMakeLists.txt#L423-L424

There are some substitutions that happen in this file, so we do need to let cmake process it before we consume it. One example: https://github.com/rstudio/rstudio/blob/88d5f9b1359d261f56a8721ca1bc03d13fbc9701/src/cpp/desktop/Info.plist.in#L270

Comment (gtritchie) [reply src/node/desktop/forge.config.js:49]: We could either create another copy of this file that lives in the electron project, and have the electron cmake file process this, or do it from the qt desktop location. Eventually we'll be removing the Qt folder altogether, but so we'll need to resolve such overlaps, but this wouldn't be the first one.

**Review by gtritchie — CHANGES_REQUESTED** [review]

File: src/node/desktop/forge.config.js (hunk 1 above)
Comment (gtritchie) [reply src/node/desktop/forge.config.js:49]: Note that we already do a bit of Electron stuff from the Qt desktop folder, see the end of the CMakelists.txt file: https://github.com/rstudio/rstudio/blob/88d5f9b1359d261f56a8721ca1bc03d13fbc9701/src/cpp/desktop/CMakeLists.txt#L681

Comment (timtmok) [reply src/node/desktop/forge.config.js:49]: Ah I missed that. Makes sense to have it go through cmake before feeding it to the Electron packaging.

Comment (timtmok) [reply src/node/desktop/forge.config.js]: It's running through CMake first now and I see that the package now has the `Info.plist` with the variables filled in.

**Comment by kevinushey** [comment]
> Listens to the open-file event which triggers for associated file open events and dragging files on to the app icon. For project files, this sets the env var before the session is created so that the project is opened.

Do you know if this will also work for Windows?

**Comment by timtmok** [comment]
> > Listens to the open-file event which triggers for associated file open events and dragging files on to the app icon. For project files, this sets the env var before the session is created so that the project is opened.
> 
> Do you know if this will also work for Windows?

No, this event is for MacOS only. However, the changes to handle the command-line arguments should also apply to Windows but I haven't tested it.

**Review by gtritchie — APPROVED** [review]
LGTM!

**Pull request merged**